### ISSUE TITLE
Staff can create, edit, delete, mark complete checklist items

### ISSUE THREAD

**Issue by kasugaijin** [rubyforgood/pet-rescue#256]
As a staff user I can add/edit/delete a checklist item.

This issue is for the backend work to ensure that we have a checklist feature that allows staff to:
- add items
- edit items
- delete items
- mark item complete
- each checklist is associated with a pet

Some of this work was started, so please review and complete it. For context, this checklist will be visible on the pets show page under the tasks tab. Preceeds https://github.com/rubyforgood/pet-rescue/issues/257

**Comment by kasugaijin** [comment]
Oops I assigned you on this one yesterday @ErinClaudio but I will unassign now as you have another ticket :)

**Comment by ErinClaudio** [comment]
I can do this one also if you would like. I am almost done with the other ticker @kasugaijin

**Comment by kasugaijin** [comment]
@ErinClaudio Ok assigned to you now.

**Comment by ErinClaudio** [comment]
@kasugaijin @edwinthinks 
The checklist will be visible on the Pets Show Page under the Tasks tab. However, I'd like to confirm if the work should be done on:
http://localhost:3000/alta/pets/5?active_tab=overview
A task partial must be created before proceeding with the checklist feature implementation. This partial would then house the checklist feature.
I look forward to your guidance. 
Thank you

**Comment by kasugaijin** [comment]
Hi @ErinClaudio 
Yes please do that! Thank you.

**Comment by ErinClaudio** [comment]
@kasugaijin Thank you

**Comment by kasugaijin** [comment]
https://github.com/all-contributors
please add @ErinClaudio for code.

**Comment by kasugaijin** [comment]
@all-contributors
please add @ErinClaudio  for code.

**Comment by rubinously** [comment]
If this is not blocked or already spoken for I can pick it up.

**Comment by phonghpham** [comment]
If this is not blocked or already spoken for I can pick it up.

**Comment by kasugaijin** [comment]
@phonghpham This issue has a PR...just waiting on some last changes. This will unblock some more work related to the tasks. 
In the meantime, are you interested in a ticket to build out filtering on the adopterion applications page? Similar to the filter on Pets Index page, but for applications so they can be filtered by pet name, applicant name, application status. Here: https://github.com/rubyforgood/pet-rescue/issues/355
